library/src/animationManager.cxx

```diff
@@ -341,26 +473,40 @@ void animationManager::UpdateForAnimationIndex()
       // Accumulate time ranges
       this->TimeRange[0] = std::min(timeRange[0], this->TimeRange[0]);
       this->TimeRange[1] = std::max(timeRange[1], this->TimeRange[1]);
+      foundAnimation = true;
     }
   }
-  if (this->TimeRange[0] > this->TimeRange[1])
+
+  if (foundAnimation)
   {
-    log::warn("Animation(s) time range delta is invalid: [", this->TimeRange[0], ", ",
-      this->TimeRange[1], "]. Swapping range.");
-    std::swap(this->TimeRange[0], this->TimeRange[1]);
+    // Check time range is valid
+    if (this->TimeRange[0] > this->TimeRange[1])
+    {
+      log::warn("Animation(s) time range delta is invalid: [", this->TimeRange[0], ", ",
```

Comment:
cannot be tested?

Reply:
No, it was not tested before and it is not tested now either. This would require to have an importer that provide incorrect temporal information. I think exodus reader could be able to provide that but I do not have a way to edit a binary exodus file.

Reply:
fair enough